### PR TITLE
Fixes #23 Extensions to #11 Programatically add tags for organs to ce…

### LINKED ISSUE #23: Extensions to #11 Programatically add tags for organs to cell types
Turn #16 into a script with args specified via that takes [neo4j2owl-config.yaml](https://github.com/kharchenkolab/cap-pipeline-config/blob/main/config/prod/neo4j2owl-config.yaml) as an input, modifies it and writes it to disk. 

Use [argparse](https://docs.python.org/3/library/argparse.html) to allow the script to be pointed to the file, rather than hardwiring it.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #23 Extensions to #11 Programatically add tags for organs to cell types

### CODE REVIEW

File: config/collectdata/python/write_organ_tags.py [hunk 1]
```diff
@@ -48,17 +63,16 @@
 yaml = ruamel.yaml.YAML()
 yaml.indent(sequence=4, offset=2)
 
-with open('neo4j2owl-config.yaml') as file:
+with open(fileName) as file:
```
Comment (ubyndr) [inline config/collectdata/python/write_organ_tags.py:66]: Can you change fileName to file_name?
https://peps.python.org/pep-0008/#function-and-variable-names

File: config/collectdata/python/write_organ_tags.py [hunk 2]
```diff
@@ -48,17 +63,16 @@
...
     yaml_config = yaml.load(file)
 
 # generate dictionary and populate with organ IRIs and labels
 # organ_labels = {"neo_node_labelling": []}
 for organ in organs:
-#   print(organ)
+    print(organ)
```
Comment (ubyndr) [inline config/collectdata/python/write_organ_tags.py:72]: I think this was used for debugging purposes, you might remove it or keep it as you wish 😄

File: config/collectdata/python/write_organ_tags.py [hunk 3]
```diff
@@ -48,17 +63,16 @@
...
     a = {'classes': organ[0], 'label': organ[1]}
     yaml_config['neo_node_labelling'].append(a)
 
 # export populated dictionary to file
-# ? can neo4j2owl-config.yaml point to this file?
-with open('neo4j2owl-config.yaml', 'w') as file:
-    documents = yaml.dump(yaml_config, file)
\ No newline at end of file
+with open(fileName, 'w') as file:
```
Comment (ubyndr) [inline config/collectdata/python/write_organ_tags.py:77]: fileName to file_name

**Review by ubyndr — CHANGES_REQUESTED** [review]
You should avoid using camel cases in python, try using underscore between the terms in variable and function names

File: config/collectdata/python/write_organ_tags.py [hunk 4]
```diff
@@ -2,13 +2,28 @@
 # generate_organ_tags.py v1.0.0
 
 # ACTION:
-#  Generate list of IRIs and labels from UBERON organ_slim and output in JSON format
+#  Generate list of IRIs and labels from UBERON organ_slim and output in JSON format to YAML file
```
Comment (evanbiederstedt) [inline config/collectdata/python/write_organ_tags.py:5]: `output in YAML format`

File: config/collectdata/python/write_organ_tags.py [hunk 5]
```diff
@@ -2,13 +2,28 @@
...
 #  Output will be used to filter and/or boost search results in Cell Annotation Platform (CAP)
 
+# INSTRUCTIONS:
+#  Organ IRIs and labels will be directed to a file named neo4j2owl-config.yaml by default
+#  To direct output to a differnt file, execute by running:
+#   python write_organ_tags.py -f file_name
+#   Replace 'file_name' with name of YAML file
```
Comment (evanbiederstedt) [inline config/collectdata/python/write_organ_tags.py:12]: By default, python's argparse allows for `--help` as a flag. This is useful here as well.

**Review by ubyndr — APPROVED** [review]

**Review by ubyndr — APPROVED** [review]

File: config/collectdata/python/write_organ_tags.py [hunk 6]
```diff
@@ -36,29 +51,29 @@
 queryOutput = []
 for line in ret["results"]["bindings"]:
     queryOutput.append(line)
-# generate list of IRIs and organ labels
+
+# generate list of organ cell DL queries and semantic labels
 organs = []
 for n in queryOutput:
     IRI = n['x']['value'].replace("_", ":")
```
Comment (dosumis) [inline config/collectdata/python/write_organ_tags.py:58]: The var name IRI is confusing here.

File: config/collectdata/python/write_organ_tags.py [hunk 7]
```diff
@@ -36,29 +51,29 @@
...
     IRI = IRI.partition('http://purl.obolibrary.org/obo/')[-1]
+    IRI = "cell and 'part of' some " + IRI
```
Comment (dosumis) [inline config/collectdata/python/write_organ_tags.py:60]: The class expression (DL query) should use curies for all entities  - cell & part_of as well as the target organ.

**Review by dosumis — CHANGES_REQUESTED** [review]
As well as changes below:  Somewhere we should have a requirements.txt file with a list of dependencies (currently just `SPARQLWrapper`).  @ubyndr - I'm wondering where this should go.  I think by convention it should live at the top level of the repo.

File: config/collectdata/python/write_organ_tags.py (hunk 6 above)
Comment (bvarner-ebi) [reply config/collectdata/python/write_organ_tags.py:58]: Good catch. Please see updated var names.

File: config/collectdata/python/write_organ_tags.py (hunk 7 above)
Comment (bvarner-ebi) [reply config/collectdata/python/write_organ_tags.py]: Updated var name

**Comment by ubyndr** [comment]
> As well as changes below: Somewhere we should have a requirements.txt file with a list of dependencies (currently just `SPARQLWrapper`). @ubyndr - I'm wondering where this should go. I think by convention it should live at the top level of the repo.

Would it be good to move the script from collectdata/script folder to script folder in the root? Then we can put the requirement.txt in there as well, so we can keep it separated from rest of the configuration logic

**Comment by bvarner-ebi** [comment]
@ubyndr, I made a 'script' directory in root and moved scripts there. Also added requirements.txt.

**Pull request merged**